src/includes/class_reservationPermissions.php

```diff
@@ -321,7 +368,9 @@ public function insertRecord($id, $type, $room, $email){
             throw new Exception("Invalid ID provided.");
         }
 
-        if (self::duplicatePermissionsCheck($id, $type, $room, $email) === FALSE) {
+        $dupCheck = self::duplicatePermissionsCheck($id, $type, $room, $email);
+
+        if ($dupCheck === FALSE) {
```

Comment:
This was sudo code, yes it should go back to the if.  During debugging this was modified to be dumped.  Also there was a static call to a private function that didn't have a static method.  So it was causing problems and never being called.